packages/types/src/type/Vote.ts

```diff
@@ -2,41 +2,98 @@
 // This software may be modified and distributed under the terms
 // of the Apache-2.0 license. See the LICENSE file for details.
 
-import { isBoolean } from '@polkadot/util';
+import { isBoolean, isNumber, isObject } from '@polkadot/util';
 
+import Conviction from './Conviction';
+import Struct from '../codec/Struct';
+import U8a from '../codec/U8a';
+import Boolean from '../primitive/Bool';
 import I8 from '../primitive/I8';
+import { AnyJsonObject } from '../types';
 
 /**
  * @name Vote
  * @description
  * A number of lock periods, plus a vote, one way or the other.
  */
-export default class Vote extends I8 {
+export default class Vote extends U8a {
+  private _aye: Boolean | I8;
+  private _conviction?: Conviction;
+
   constructor (value?: any) {
-    super(Vote.decodeVote(value));
+    const decoded = Vote.decodeVote(value);
```

Comment:
in `decodeVote`, it will be much simpler to always return a Uint8Array. then below, you don't need to distinguish the cases between the 2 decoded, just `super(decoded)` always.